src/Microsoft.ManagedZlib/src/Deflater.cs

```diff
@@ -0,0 +1,273 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Security;
+
+using ZErrorCode = Microsoft.ManagedZLib.ManagedZLib.ErrorCode;
+using ZFlushCode = Microsoft.ManagedZLib.ManagedZLib.FlushCode;
+
+namespace Microsoft.ManagedZLib
+{
+    /// <summary>
+    /// Provides a wrapper around the ZLib compression API.
+    /// </summary>
+    internal sealed class Deflater : IDisposable
```

Comment:
I suggested to keep `IDisposable` in case we decide to use `ArrayPool` for the `InputBuffer` and `OutputBuffer` types that are used in the `Inflater`, at which point we'd want to be able to dispose them (which would eventually propagate up to here). I didn't realize that the `IDisposable` was only on the `Deflater`.

Reply:
It was on both, but I didn't end up using `ArrayPool` in `Inflater`. `Deflater` is still in progress.